zipline/utils/serialization_utils.py

```diff
@@ -13,6 +13,66 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+from six import BytesIO
+import pickle
+from functools import partial
+
+from zipline.assets import AssetFinder
+from zipline.finance.trading import TradingEnvironment
+
 # Label for the serialization version field in the state returned by
 # __getstate__.
 VERSION_LABEL = '_stateversion_'
+
+
+def _persistent_id(obj):
+    if isinstance(obj, AssetFinder):
+        return AssetFinder.PERSISTENT_TOKEN
+    if isinstance(obj, TradingEnvironment):
+        return TradingEnvironment.PERSISTENT_TOKEN
+    return None
+
+
+def _persistent_load(persid, env):
+    if persid == AssetFinder.PERSISTENT_TOKEN:
+        return env.asset_finder
+    if persid == TradingEnvironment.PERSISTENT_TOKEN:
+        return env
+
+
+def dump_with_persistent_ids(obj, protocol=None):
```

Comment:
@richafrank This is the method we discussed that can be used in place of pickle.dumps and performs the tokenization of the AssetFinder and TradingEnvironment.

There is a corresponding method below to be used in place of pick.load.